examples/mint-repl/mint-repl.rs

```diff
@@ -878,62 +773,160 @@ fn prepare_reissue_cli(mintinfo: &mut MintInfo) -> Result<()> {
 }
 
 /// Implements reissue command.
-fn reissue_cli(mintinfo: &mut MintInfo) -> Result<()> {
+// Not necessary until multisig Dbc owner is supported
+#[allow(dead_code)]
+fn reissue_prepared_cli(mintinfo: &mut MintInfo) -> Result<()> {
     let mr_input = readline_prompt_nl("\nReissueRequest: ")?;
     let reissue_request: ReissueRequestUnblinded = from_be_hex(&mr_input)?;
 
     println!("\n\nThank-you.   Generating DBC(s)...\n\n");
 
-    reissue(
-        mintinfo,
-        &reissue_request.inner,
-        &reissue_request.output_pk_pks,
-    )
+    reissue(mintinfo, reissue_request)
 }
 
-/// Implements reissue_ez command.
-fn reissue_ez(mintinfo: &mut MintInfo) -> Result<()> {
-    let tx = prepare_tx()?;
-    let sig_map = sign_tx(&tx)?;
-    let rr = prepare_reissue(mintinfo, tx, sig_map)?;
-    reissue(mintinfo, &rr.inner, &rr.output_pk_pks)
+struct ReissueAuto {
+    pub(crate) unspent_dbcs: HashMap<[u8; 32], Dbc>,
+}
+
+impl From<Vec<Dbc>> for ReissueAuto {
+    fn from(unspent_dbcs: Vec<Dbc>) -> Self {
+        Self {
+            unspent_dbcs: HashMap::from_iter(unspent_dbcs.into_iter().map(|d| (d.hash(), d))),
+        }
+    }
+}
+
+fn reissue_auto_cli(mintinfo: &mut MintInfo) -> Result<()> {
+    let mut rng = rand::thread_rng();
+    let mut rng8 = rand8::thread_rng();
+
+    let num_reissues: usize =
+        readline_prompt_nl_default("\nHow many reissues to perform [10]: ", "10")?.parse()?;
+
+    for _i in 1..=num_reissues {
+        let max_inputs = vec![mintinfo.reissue_auto.unspent_dbcs.len(), 10]
+            .into_iter()
+            .min()
+            .unwrap_or(10);
+        let num_inputs = if 1 == max_inputs {
+            1
+        } else {
+            rng.gen_range(1, max_inputs)
+        };
+
+        // subset of unspent_dbcs become the inputs for next reissue.
+        let input_dbcs: Vec<Dbc> = mintinfo
+            .reissue_auto
+            .unspent_dbcs
+            .iter()
+            .choose_multiple(&mut rng, num_inputs)
+            .iter()
+            .map(|(_, d)| (*d).clone())
+            .collect();
+
+        let mut tx_builder = TransactionBuilder::default();
+
+        for dbc in input_dbcs.iter() {
+            let base_sk = dbc.owner_base().secret_key()?;
+            let decoy_inputs = mintinfo.spentbook()?.random_decoys(STD_DECOYS, &mut rng8);
+
+            tx_builder = tx_builder.add_input_dbc(dbc, &base_sk, decoy_inputs, &mut rng8)?;
+
+            mintinfo.reissue_auto.unspent_dbcs.remove(&dbc.hash());
+        }
+
+        let inputs_sum = tx_builder.inputs_amount_sum();
+
+        while tx_builder.outputs_amount_sum() < inputs_sum {
+            // randomize output amount
+            let diff = inputs_sum - tx_builder.outputs_amount_sum();
+            let amount = if diff == 1 { 1 } else { rng.gen_range(1, diff) };
```

Comment:
done.  though I had to modify the while loop condition like so:
```
while tx_builder.outputs_amount_sum() < inputs_sum || tx_builder.outputs().is_empty() {
```
otherwise zero valued outputs can become inputs which sum to zero, so the while body does not exec, leading to an invalid tx with zero outputs.

that's a very subtle one.